api/interface.go

```diff
@@ -327,14 +327,18 @@ type Connection interface {
 	// Close closes the connection.
 	Close() error
 
-	// Addr returns the address used to connect to the API server.
-	Addr() string
+	// Addr returns a copy of the address used to connect to the API server.
+	Addr() *url.URL
 
 	// IPAddr returns the IP address used to connect to the API server.
 	IPAddr() string
 
 	// APIHostPorts returns addresses that may be used to connect
-	// to the API server, including the address used to connect.
+	// to the API server, excluding the address used to connect.
+	// Use Addr() and IsProxied() to inspect the address used to
+	// to connect. This distinction is made because HostPorts
+	// do not carry information for routing through things like
+	// L7 load-balancers while Addr() does.
```

Comment:
The behaviour of this method (`APIHostPorts()`) has changed. I've checked for usage in Juju and the only one is in `juju/api.go` which has been updated to take this new behaviour into account. External consumers of the Connection interface will experience a breaking change however if they really on `APIHostPorts()`.